clients/http_json_rpc_client.go

```diff
@@ -181,6 +193,14 @@ func (c *GenericHttpJsonRpcClient) shutdown() {
 	c.processBatch(true)
 }
 
+func (c *GenericHttpJsonRpcClient) getHttpClient() *http.Client {
+	if c.proxyPool != nil && len(c.proxyPool.clients) > 0 {
+		return c.proxyPool.GetClient()
```

Comment:
Probably it's better if internals are not exposed here, so GetClient() can return nil and we check that (vs `len(clients)` which is an internal detail)